LiteCore/Database/DatabaseImpl.cc

```diff
@@ -762,6 +768,17 @@ namespace litecore {
     }
 
 
+    void DatabaseImpl::collectionRemoved(slice scope, slice name) {
+        LOCK(_collectionsMutex);
+        auto c = _collections.find({name, scope});
+        if(c != _collections.end()) {
+            CollectionImpl* coll = (CollectionImpl *)c->second.get();
+            coll->close();
```

Comment:
C4CollectionImpl is not thread-safe, and calling `close()` from another thread can cause a number of Bad Things, if the regular thread is using that state:
- `_documentFactory` could be freed while another thread is in the middle of calling it to create a document.
- `_sequenceTracker` could be freed while another thread is calling it while trying to start a transaction.
- `_housekeeper` could be released and freed while another thread is updating doc expiration or expiring docs.

You could add a mutex to CollectionImpl to handle this, but you'd have to add a lot of lock calls.

This is why I was preferring not to do this kind of notification -- instead either just handling the no-such-table errors from SQLite, or preflighting.